### PR TITLE
Added tips

### PR DESCRIPTION
**Pull request by beaufortfrancois**
As discussed in https://github.com/mounirlamouri/manifest-validator/pull/9, here's a simple way to show common tips when `JSON.parse` fails.

### CODE REVIEW

File: manifest-parser.js [hunk 1]
```diff
@@ -224,12 +225,15 @@ var ManifestParser = (function() {
...
     // TODO: temporary while ManifestParser is a collection of static methods.
     _logs = [];
+    _tips = [];
     _success = true;
 
     try {
       _json_input = JSON.parse(string);
     } catch (e) {
       _logs.push("File isn't valid JSON: " + e);
+      _tips.push("Double quotes are considered valid JSON, single quotes aren't. Make sure you use double quotes.");
```
Comment (mounirlamouri) [inline manifest-parser.js]: I would do something like:

``` js
_tips.push('Your JSON failed to parse, these are the main reasons why JSON parsing usually fails:\n - Double quotes should be used around property names and for strings. Single quotes are not valid.\n - JSON specification disallow trailing comma after the last property even if some implementations allow it.');
```

IOW:
- mention that the tips are related to the JSON parsing failure.
- mention when double quotes are usually used.
- use single quotes ;)
- optional: make this only one tip?

lgtm with this.

Comment (beaufortfrancois) [inline manifest-parser.js]: Done.

**Comment by beaufortfrancois** [comment]
Your feedback has been addressed.

File: manifest-parser.js [hunk 2]
```diff
@@ -224,12 +225,14 @@ var ManifestParser = (function() {
...
     // TODO: temporary while ManifestParser is a collection of static methods.
     _logs = [];
+    _tips = [];
     _success = true;
 
     try {
       _json_input = JSON.parse(string);
     } catch (e) {
       _logs.push("File isn't valid JSON: " + e);
+      _tips.push('Your JSON failed to parse, these are the main reasons why JSON parsing usually fails:\n - Double quotes should be used around property names and for strings. Single quotes are not valid.\n - JSON specification disallow trailing comma after the last property even if some implementations allow it.');
```
Comment (samccone) [inline manifest-parser.js]: can we break this string into multiple lines < 80chars ... otherwise this will fail on the jscs style lint

Comment (mounirlamouri) [inline manifest-parser.js]: François, can you do that, rebase and merge?

Comment (beaufortfrancois) [inline manifest-parser.js]: giphy

**Pull request merged**